// python -m grpc_tools.protoc -I protos --python_out=. --pyi_out=. --grpc_python_out=. protos/chat.proto

syntax = "proto3";

package chatservice;

service AuthService {
  rpc Login (LoginRequest) returns (LoginResponse) {}
  rpc Register (RegisterRequest) returns (RegisterResponse) {}
}

service ChatService {
  rpc SendMessage (SendRequest) returns (SendResponse) {}
  rpc ReceiveMessage (User) returns (stream ChatMessage) {}
  rpc GetUsers (Empty) returns (stream User) {}
  // rpc Receive
}

message Empty {}

message User{
  string username = 1;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string message = 2;
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
}

message SendRequest {
  string sender = 1;
  string receiver = 2;
  string body = 3;
}

message SendResponse {
  bool success = 1;
  string message = 2;
}

message ChatMessage {
  string sender = 1;
  string body = 2;
}